### PR TITLE
Fixed #30595 -- Added error message when no objects found to sql* management commands.

### PR DESCRIPTION
**Pull request by hramezani**
[ticket 30595](https://code.djangoproject.com/ticket/30595)

### CODE REVIEW

File: django/core/management/commands/sqlflush.py [hunk 1]
```diff
@@ -19,4 +19,7 @@ def add_arguments(self, parser):
         )
 
     def handle(self, **options):
-        return '\n'.join(sql_flush(self.style, connections[options['database']], only_django=True))
+        sql_statements = sql_flush(self.style, connections[options['database']], only_django=True)
+        if not sql_statements and options['verbosity'] >= 1:
+            self.stderr.write('-- No statements found.')
```
Comment (felixxm) [inline django/core/management/commands/sqlflush.py]: We can remove `-- ` prefix from all outputs.

File: tests/migrations/test_commands.py [hunk 2]
```diff
@@ -645,6 +645,12 @@ def test_sqlmigrate_for_non_transactional_databases(self):
             self.assertNotIn(start_transaction_sql.lower(), queries)
         self.assertNotIn(connection.ops.end_transaction_sql().lower(), queries)
 
+    @override_settings(MIGRATION_MODULES={'migrations': 'migrations.test_migrations_sqlmigrate'})
```
Comment (felixxm) [inline tests/migrations/test_commands.py]: Maybe `test_migrations_sqlmigrate` -> `test_migrations_no_operations`

File: tests/migrations/test_commands.py [hunk 3]
```diff
@@ -645,6 +645,12 @@ def test_sqlmigrate_for_non_transactional_databases(self):
...
+    def test_sqlmigrate_no_statements(self):
+        err = io.StringIO()
+        call_command('sqlmigrate', 'migrations', '0001_initial', stderr=err)
+        self.assertEqual(err.getvalue().strip(), 'No statements found.')
```
Comment (felixxm) [inline tests/migrations/test_commands.py]:
```suggestion
        self.assertEqual(err.getvalue(), 'No statements found.\n')
```

File: tests/migrations/test_migrations_sqlmigrate/0001_initial.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+    ]
+
+    operations = [
+    ]
```
Comment (felixxm) [inline tests/migrations/test_migrations_sqlmigrate/0001_initial.py]: Please minimize this file, e.g.
```python
class Migration(migrations.Migration):
    dependencies = []
    operations = []
```

File: tests/backends/base/test_operations.py [hunk 5]
```diff
@@ -41,6 +44,18 @@ def test_sql_flush(self):
...
 
+    def test_sqlflush_no_statements(self):
+        err = io.StringIO()
+        with mock.patch('django.core.management.sql.sql_flush', return_value=''):
+            call_command('sqlflush', stderr=err)
+            self.assertEqual(err.getvalue().strip(), 'No statements found.')
+
+    def test_sqlsequencereset_no_sequences(self):
+        err = io.StringIO()
+        with mock.patch('django.db.connection.ops.sequence_reset_sql', return_value=''):
```
Comment (felixxm) [inline tests/backends/base/test_operations.py]: It should return an empty list (the same in `sqlflush`).

File: django/core/management/commands/sqlmigrate.py [hunk 6]
```diff
@@ -63,4 +63,6 @@ def handle(self, *args, **options):
         # for it
         plan = [(executor.loader.graph.nodes[targets[0]], options['backwards'])]
         sql_statements = executor.collect_sql(plan)
+        if not sql_statements and options['verbosity'] >= 1:
+            self.stderr.write('No statements found.')
```
Comment (felixxm) [inline django/core/management/commands/sqlmigrate.py]: Maybe `statements` -> `operations`.

File: django/core/management/commands/sqlflush.py [hunk 7]
```diff
@@ -19,4 +19,7 @@ def add_arguments(self, parser):
         )
 
     def handle(self, **options):
-        return '\n'.join(sql_flush(self.style, connections[options['database']], only_django=True))
+        sql_statements = sql_flush(self.style, connections[options['database']], only_django=True)
+        if not sql_statements and options['verbosity'] >= 1:
+            self.stderr.write('No statements found.')
```
Comment (felixxm) [inline django/core/management/commands/sqlflush.py]: `statements` -> `tables`?

**Comment by hramezani** [comment]
@felixxm Thanks for the comments. Done!

**Comment by felixxm** [comment]
@hramezani Thanks! I rebased from master and moved tests to a new directory `empty_models` because we don't test backends' operations but core commands for apps without models.

**Pull request merged**